re: the failing pipeline:

```
       > internal/prometheus/prometheus.go:165:6: func boolToString is unused (unused)
       > func boolToString(b bool) string {
       >      ^
       > 1 issues:
       > * unused: 1
```

the function boolToString was already there, it's now unused but may be nice to have in the future, should I remove it for now?

--- a/internal/prometheus/prometheus.go
+++ b/internal/prometheus/prometheus.go
@@ -132,7 +132,7 @@ func updateFetched(fetched *protobuf.Event_Fetched, metrics *Prometheus) {
 		}
 
 		metrics.IncFetchCounter(repo.GetName(), status)
-		metrics.lastFetchFailed.With(prometheus.Labels{"remote_name": remoteName}).Set(boolToFloat64(!success))
+		metrics.lastFetchFailed.With(prometheus.Labels{"remote_name": repo.GetName()}).Set(boolToFloat64(!success))
 	}
 }
 

--- a/internal/prometheus/prometheus.go
+++ b/internal/prometheus/prometheus.go
@@ -162,13 +162,6 @@ func (m Prometheus) SetLastFetchFailed(remoteName string, failed bool) {
 	m.lastFetchFailed.With(prometheus.Labels{"remote_name": remoteName}).Set(boolToFloat64(failed))
 }
 
-func boolToString(b bool) string {
-	if b {
-		return "1"
-	}
-	return "0"
-}
-
 func boolToFloat64(b bool) float64 {
 	if b {
 		return 1

--- a/internal/prometheus/prometheus.go
+++ b/internal/prometheus/prometheus.go
@@ -123,6 +123,7 @@ func Subscribe(broker *brokerPkg.Broker, metrics *Prometheus) {
 }
 
 func updateFetched(fetched *protobuf.Event_Fetched, metrics *Prometheus) {
+	metrics.lastFetchFailed.Reset()
 	for _, repo := range fetched.RepositoryStatus.GetRemotes() {
 		status := "failed"
 		success := repo.GetFetched().GetValue()
@@ -131,7 +132,7 @@ func updateFetched(fetched *protobuf.Event_Fetched, metrics *Prometheus) {
 		}
 
 		metrics.IncFetchCounter(repo.GetName(), status)
-		metrics.SetLastFetchFailed(repo.GetName(), !success)
+		metrics.lastFetchFailed.With(prometheus.Labels{"remote_name": remoteName}).Set(boolToFloat64(!success))
 	}
 }
 
@@ -155,11 +156,6 @@ func (m Prometheus) SetBuildInfo(version string) {
 func (m Prometheus) SetDeploymentInfo(commitId, status string) {
 	m.deploymentInfo.Reset()
 	m.deploymentInfo.With(prometheus.Labels{"commit_id": commitId, "status": status}).Set(1)
-}
-
-func (m Prometheus) SetLastFetchFailed(remoteName string, failed bool) {
-	m.lastFetchFailed.Reset()
-	m.lastFetchFailed.With(prometheus.Labels{"remote_name": remoteName}).Set(boolToFloat64(failed))
 }
 
 func boolToFloat64(b bool) float64 {

